--- a/net/badlion/client/session/BadlionSession.java
+++ b/net/badlion/client/session/BadlionSession.java
@@ -12,11 +12,14 @@
    }
 
    public BadlionSession(String var1, String var2, String var3, String var4, String var5) {
-      this.field_2134 = new Session(var1, var2, var3, var4, var5);
+//      this.field_2134 = new Session(var1, var2, var3, var4, var5);
+      //TODO: mixin fix
+      this.field_2134 = new Session(var1, var2, var3, var4);
    }
 
    public void method_17(String var1) {
-      this.field_2134.setToken(var1);
+      //TODO: mixin fix
+//      this.field_2134.setToken(var1);
    }
 
    public String method_16() {
@@ -28,7 +31,9 @@
    }
 
    public UUID getPlayerUuid() {
-      return this.field_2134.getPlayerUUID();
+      //TODO: mixin fix
+      throw new RuntimeException("getPlayerUuid not implemented");
+//      return this.field_2134.getPlayerUUID();
    }
 
    public String method_20() {
@@ -44,6 +49,8 @@
    }
 
    public String method_23() {
-      return this.field_2134.getUserId();
+      //TODO: mixin fix
+      throw new RuntimeException("method_23 not implemented");
+//      return this.field_2134.getUserId();
    }
 }
