--- a/net\badlion\client\common\util\Colour.java
+++ b/net\badlion\client\common\util\Colour.java
@@ -113,9 +113,12 @@
         if (!this.enabled) {
             return Class_2125.method_5865(this.method_2788());
         } else {
-            return this.field_1752 != null ? this.field_1752 : this.field_1752 = Class_2125.method_5865(this.color);
+            if(field_1752 == null){
+                this.field_1752 = Class_2125.method_5865(this.color);
         }
+            return this.field_1752;
     }
+    }
 
     public void method_2783(int parameter_1) {
         this.color = new net.badlion.client.common.colour.Colour(Class_2125.method_5867(parameter_1), Class_2125.method_5869(parameter_1), Class_2125.method_5870(parameter_1), Class_2125.method_5871(parameter_1));
