--- /dev/null
+++ b/io/github/badlionmoddinggroup/badlion/gui/BadlionButtonWidget.java
@@ -0,0 +1,21 @@
+package io.github.badlionmoddinggroup.badlion.gui;
+
+import net.minecraft.client.gui.widget.ButtonWidget;
+
+public class BadlionButtonWidget extends ButtonWidget {
+    public BadlionButtonWidget(int i, int i1, int i2, byte var2, int i3, String s, ButtonType thickLines) {
+        super(i, i1, i2, var2, i3, s);
+    }
+
+    public BadlionButtonWidget(int i, int i1, int i2, int var2, int i3, String s, ButtonType thickLines) {
+        super(i, i1, i2, var2, i3, s);
+    }
+
+    public BadlionButtonWidget(int i, int i1, int i2, byte var2, int i3, String s) {
+        super(i, i1, i2, var2, i3, s);
+    }
+
+    public BadlionButtonWidget(int i, int i1, int i2, int var2, int i3, String s) {
+        super(i, i1, i2, var2, i3, s);
+    }
+}
