--- a/net/badlion/input/CursorHandler.java
+++ b/net/badlion/input/CursorHandler.java
@@ -1,12 +1,14 @@
 package net.badlion.input;
 
 import java.awt.image.BufferedImage;
+
+import org.lwjgl.LWJGLException;
 import org.lwjgl.input.Cursor;
 
 public interface CursorHandler {
    void method_4494(CursorHandler.class_1636 var1);
 
-   Cursor method_4493(BufferedImage var1);
+   Cursor method_4493(BufferedImage var1) throws LWJGLException;
 
    boolean method_4495(CursorHandler.class_1636 var1);
 
