--- a/net/badlion/client/mods/replaymod/ReplayManager.java
+++ b/net/badlion/client/mods/replaymod/ReplayManager.java
@@ -59,11 +59,12 @@
          PacketByteBuf var4 = new PacketByteBuf(var3);
          int var5 = var4.readVarInt();
          Packet var6 = null;
-         if (var2) {
-            var6 = NetworkState.field_5980.method_5123(NetworkSide.field_6044, var5);
-         } else {
-            var6 = NetworkState.field_5978.method_5123(NetworkSide.field_6044, var5);
-         }
+         //TODO unknown fix
+//         if (var2) {
+//            var6 = NetworkState.field_5980.method_5123(NetworkSide.CLIENTBOUND, var5);
+//         } else {
+//            var6 = NetworkState.field_5978.method_5123(NetworkSide.CLIENTBOUND, var5);
+//         }
 
          var6.read(var4);
          class_1116 var7 = new class_1116(var6);
@@ -90,7 +91,7 @@
 
    }
 
-   protected void method_5289(ConcurrentLinkedQueue var1) {
+   public void method_5289(ConcurrentLinkedQueue var1) {
       var1.add(new class_435(0L, new class_1116(new LoginSuccessS2CPacket(new GameProfile(UUID.nameUUIDFromBytes(new byte[16]), "Player")))));
    }
 
