--- a/net/badlion/client/common/mod/ChunkBordersMod.java
+++ b/net/badlion/client/common/mod/ChunkBordersMod.java
@@ -25,7 +25,7 @@
 
 public class ChunkBordersMod extends AbstractMod {
    private static final KeybindWrapper field_916;
-   private static final Keyboard.Keybind field_917;
+   private static Keyboard.Keybind field_917 = null;
    @KeepName
    private Keyboard.Keybind toggleKey;
    @KeepName
@@ -289,7 +289,7 @@
    }
 
    public void method_1834(ChunkBordersMod.class_562 var1) {
-      GL11.glLineWidth((float)var1.method_1838().getValue());
+      GL11.glLineWidth(var1.method_1838().getValue().floatValue());
       this.field_919 = var1.method_1836();
       this.field_918 = var1.method_1836().method_2781() == Colour.class_952.CHROMA;
    }
