--- a/net\badlion\client\common\gui\ClientGui.java
+++ b/net\badlion\client\common\gui\ClientGui.java
@@ -1,6 +1,8 @@
 package net.badlion.client.common.gui;
 
 import java.util.List;
+
+import io.github.badlionmoddinggroup.badlion.util.BadlionDrawableHelper;
 import net.badlion.Class_2066;
 import net.badlion.Class_22;
 import net.badlion.Class_69;
@@ -17,7 +19,7 @@
 
     public Class_69 method_145() {
         if (this.field_1546 == null) {
-            this.field_1546 = new Class_821(DrawableHelper.backgroundPanorama);
+            this.field_1546 = new Class_821(BadlionDrawableHelper.backgroundPanorama);
         }
 
         return this.field_1546;
@@ -57,7 +59,7 @@
 
     public void drawColouredrect(int parameter_1, int parameter_2, int parameter_3, int parameter_4, Colour parameter_5) {
         if (parameter_5.method_2781() == Colour.ColorMode.CHROMA) {
-            DrawableHelper.drawChromaRect(parameter_1, parameter_2, parameter_3, parameter_4);
+            BadlionDrawableHelper.drawChromaRect(parameter_1, parameter_2, parameter_3, parameter_4);
         } else {
             this.method_150(parameter_1, parameter_2, parameter_3, parameter_4, parameter_5.method_2792());
         }
