--- a/net/badlion/client/render/font/BadlionFontRenderer.java
+++ b/net/badlion/client/render/font/BadlionFontRenderer.java
@@ -129,7 +129,8 @@
 
    public float method_5751(int var1) {
       if (BadlionTransformer.OPTIFINE_LOADED) {
-         return class_2025.OPTIFINE_VERSION.equals("L5") ? (float)MinecraftClient.getInstance().textRenderer.field_2818[var1] : MinecraftClient.getInstance().textRenderer.d[var1];
+         //FIXME: multiple fixes? needed for this one
+         return class_2025.OPTIFINE_VERSION.equals("L5") ? (float)MinecraftClient.getInstance().textRenderer.field_2818[var1] : 0;//MinecraftClient.getInstance().textRenderer.d[var1]; what the fuck is textRenderer.d[var1]?
       } else {
          return (float)MinecraftClient.getInstance().textRenderer.field_2818[var1];
       }
