--- a/net/badlion/class_1116.java
+++ b/net/badlion/class_1116.java
@@ -32,7 +32,7 @@
          var1 = Unpooled.buffer();
          PacketByteBuf var2 = new PacketByteBuf(var1);
          NetworkState var3 = NetworkState.getPacketHandlerState(this.field_2151);
-         int var4 = var3.method_5124(NetworkSide.field_6044, this.field_2151);
+         int var4 = var3.method_5124(NetworkSide.CLIENTBOUND, this.field_2151);
          var2.writeVarInt(var4);
          this.field_2151.write(var2);
          var1.readerIndex(0);
