--- a/net/badlion/class_369.java
+++ b/net/badlion/class_369.java
@@ -30,7 +30,7 @@
    private BufferedInputStream field_551;
    private boolean field_552 = false;
 
-   public class_369(File var1) {
+   public class_369(File var1) throws IOException {
       this.field_549 = var1;
       this.field_550 = new ZipFile(var1);
       this.field_551 = new BufferedInputStream(this.field_550.getInputStream(this.method_1367("recording.tmcpr")));
@@ -40,7 +40,7 @@
       return this.field_550.getEntry(var1);
    }
 
-   public BufferedInputStream method_1369(String var1) {
+   public BufferedInputStream method_1369(String var1) throws IOException {
       ZipEntry var2 = this.method_1367(var1);
       return var2 == null ? null : new BufferedInputStream(this.field_550.getInputStream(var2));
    }
@@ -208,8 +208,7 @@
          try {
             String var3;
             if (var1 == null) {
-               var3 = null;
-               return var3;
+               return null;
             }
 
             var3 = new String(ByteStreams.toByteArray(var1), Charsets.UTF_8);
@@ -253,8 +252,7 @@
          try {
             class_292 var3;
             if (var1 == null) {
-               var3 = null;
-               return var3;
+               return null;
             }
 
             var3 = new class_292();
@@ -307,7 +305,7 @@
                   try {
                      if (var4 == null) {
                         var6 = null;
-                        return var6;
+                        return null;
                      }
 
                      var3 = (JsonObject)BadlionClient.UGLY_GSON.fromJson(new InputStreamReader(var4), JsonObject.class);
@@ -422,7 +420,7 @@
 
    }
 
-   public void method_1379() {
+   public void method_1379() throws IOException {
       this.field_551.close();
       this.field_550.close();
    }
