--- a/net/badlion/class_305.java
+++ b/net/badlion/class_305.java
@@ -28,7 +28,7 @@
    private final Map field_389 = new HashMap();
    private final Set field_390 = new HashSet();
 
-   void method_1135(NickHiderMod var1) {
+   public void method_1135(NickHiderMod var1) {
       class_27 var2 = BadlionClient.minecraft.method_237();
       if (var2 != null && var2.method_5259()) {
          class_8 var3 = var2.method_5260();
@@ -155,7 +155,7 @@
       }
    }
 
-   void method_1132() {
+   public void method_1132() {
       this.field_388.clear();
       this.field_389.clear();
       this.field_390.clear();
