--- a/net\badlion\Class_248.java
+++ b/net\badlion\Class_248.java
@@ -12,6 +12,11 @@
         return new JsonPrimitive(parameter_1.method_752());
     }
 
+    @Override
+    protected Class_187 method_5282(JsonElement parameter_1) {
+        return method_966(parameter_1);
+    }
+
     public JsonElement method_5283(Class_187 parameter_1) {
         return this.method_967((Class_187)parameter_1);
     }
