--- a/net/badlion/class_1258.java
+++ b/net/badlion/class_1258.java
@@ -1,5 +1,6 @@
 package net.badlion;
 
+import java.io.IOException;
 import java.io.InputStream;
 import java.nio.ByteBuffer;
 import net.minecraft.client.resource.metadata.TextureResourceMetadata;
@@ -26,7 +27,7 @@
         this.field_2415 = var1;
     }
 
-    public void method_3456(ResourceManager var1) {
+   public void method_3456(ResourceManager var1) throws IOException {
         InputStream var2 = null;
 
         try {
