--- a/net/badlion/class_707.java
+++ b/net/badlion/class_707.java
@@ -5,7 +5,7 @@
 import net.badlion.client.common.item.tag.ITag;
 
 public class class_707 {
-    public static class_325 method_2190(class_325 var0, int var1, int var2, int var3) {
+   public static class_325 method_2190(class_325 var0, int var1, int var2, int var3) throws class_706 {
         if (var0 == null) {
             return null;
         } else {
@@ -21,7 +21,7 @@
         }
     }
 
-    public static EntityWrapper method_2189(EntityWrapper var0, int var1, int var2, int var3) {
+   public static EntityWrapper method_2189(EntityWrapper var0, int var1, int var2, int var3) throws class_706 {
         if (var0 == null) {
             return null;
         } else {
