--- a/net/badlion/class_813.java
+++ b/net/badlion/class_813.java
@@ -2,6 +2,8 @@
 
 import java.util.HashMap;
 import java.util.Map;
+
+import io.github.badlionmoddinggroup.badlion.extra.ContainerExtra;
 import net.badlion.client.common.container.ContainerWrapper;
 import net.badlion.client.common.gui.lib.GuiType;
 import net.minecraft.client.gui.TwitchErrorScreen;
@@ -62,7 +64,7 @@
     public class_813(Screen var1) {
         this.field_2110 = var1;
         if (field_1554.containsKey(var1.getClass())) {
-            this.field_1556 = (GuiType)field_1554.get(var1.getClass());
+            this.field_1556 = field_1554.get(var1.getClass());
         } else {
             this.field_1556 = GuiType.UNKNOWN;
         }
@@ -109,7 +111,7 @@
     }
 
     public ContainerWrapper method_5263() {
-        return this.field_2110 instanceof ChestScreen ? ((ChestScreen)this.field_2110).container.getBadlionContainer() : null;
+        return this.field_2110 instanceof ChestScreen ? ((ContainerExtra)((ChestScreen)this.field_2110).container).getBadlionContainer() : null;
     }
 
     public boolean method_5264() {
