--- a/net/badlion/class_767.java
+++ b/net/badlion/class_767.java
@@ -3,6 +3,8 @@
 import java.util.ArrayList;
 import java.util.LinkedList;
 import java.util.List;
+
+import io.github.badlionmoddinggroup.badlion.extra.ItemStackExtra;
 import net.badlion.client.common.item.ItemStackWrapper;
 import net.minecraft.entity.player.PlayerInventory;
 import net.minecraft.item.FoodItem;
@@ -26,7 +28,7 @@
       for(int var4 = 0; var4 < var3; ++var4) {
          ItemStack var5 = var2[var4];
          if (var5 != null && var5.getItem().equals(Items.ARROW)) {
-            var1.add(var5.getBadlionItemStack());
+            var1.add(((ItemStackExtra)(Object)var5).getBadlionItemStack());
          }
       }
 
@@ -41,7 +43,7 @@
       for(int var4 = 0; var4 < var3; ++var4) {
          ItemStack var5 = var2[var4];
          if (var5 != null && var5.getItem().equals(Items.GOLDEN_APPLE) && var5.getDamage() == 0) {
-            var1.add(var5.getBadlionItemStack());
+            var1.add(((ItemStackExtra)(Object)var5).getBadlionItemStack());
          }
       }
 
@@ -56,7 +58,7 @@
       for(int var4 = 0; var4 < var3; ++var4) {
          ItemStack var5 = var2[var4];
          if (var5 != null && var5.getItem().equals(Items.GOLDEN_APPLE) && var5.getDamage() == 1) {
-            var1.add(var5.getBadlionItemStack());
+            var1.add(((ItemStackExtra)(Object)var5).getBadlionItemStack());
          }
       }
 
@@ -71,7 +73,7 @@
       for(int var4 = 0; var4 < var3; ++var4) {
          ItemStack var5 = var2[var4];
          if (var5 != null && (var5.getItem() instanceof FoodItem || var5.getItem() == Items.CAKE)) {
-            var1.add(var5.getBadlionItemStack());
+            var1.add(((ItemStackExtra)(Object)var5).getBadlionItemStack());
          }
       }
 
@@ -86,7 +88,7 @@
       for(int var4 = 0; var4 < var3; ++var4) {
          ItemStack var5 = var2[var4];
          if (var5 != null && var5.getItem() instanceof PotionItem) {
-            var1.add(var5.getBadlionItemStack());
+            var1.add(((ItemStackExtra)(Object)var5).getBadlionItemStack());
          }
       }
 
@@ -101,7 +103,7 @@
       for(int var4 = 0; var4 < var3; ++var4) {
          ItemStack var5 = var2[var4];
          if (var5 != null && var5.getItem() instanceof StewItem) {
-            var1.add(var5.getBadlionItemStack());
+            var1.add(((ItemStackExtra)(Object)var5).getBadlionItemStack());
          }
       }
 
@@ -119,7 +121,7 @@
          if (var6 == null) {
             var2.add((Object)null);
          } else {
-            var2.add(var6.getBadlionItemStack());
+            var2.add(((ItemStackExtra)(Object)var6).getBadlionItemStack());
          }
       }
 
@@ -128,7 +130,7 @@
 
    public ItemStackWrapper method_1947() {
       ItemStack var1 = this.field_2074.getMainHandStack();
-      return var1 == null ? null : var1.getBadlionItemStack();
+      return var1 == null ? null : ((ItemStackExtra)(Object)var1).getBadlionItemStack();
    }
 
    public int method_1948() {
@@ -141,7 +143,7 @@
 
       for(int var3 = 0; var3 < var1; ++var3) {
          ItemStack var4 = this.field_2074.main[var3];
-         var2[var3] = var4 == null ? null : var4.getBadlionItemStack();
+         var2[var3] = var4 == null ? null : ((ItemStackExtra)(Object)var4).getBadlionItemStack();
       }
 
       return var2;
@@ -149,6 +151,6 @@
 
    public ItemStackWrapper method_1940(int var1) {
       ItemStack var2 = this.field_2074.armor[var1];
-      return var2 == null ? null : var2.getBadlionItemStack();
+      return var2 == null ? null : ((ItemStackExtra)(Object)var2).getBadlionItemStack();
    }
 }
