--- a/net/badlion/class_2144.java
+++ b/net/badlion/class_2144.java
@@ -8,10 +8,7 @@
 import com.google.common.collect.Sets;
 import com.google.gson.JsonParseException;
 import java.awt.image.BufferedImage;
-import java.io.FileNotFoundException;
-import java.io.InputStreamReader;
-import java.io.Reader;
-import java.io.StringReader;
+import java.io.*;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
@@ -40,7 +37,7 @@
 public class class_2144 {
    private static final Vector3f field_5840 = new Vector3f(0.0F, 0.0F, 0.0F);
    private static final Vector3f field_5841 = new Vector3f(16.0F, 16.0F, 16.0F);
-   private static final List field_5842 = Lists.newArrayList();
+   private static final List<ModelIdentifier> field_5842 = Lists.newArrayList();
    private static final ListMultimap field_5843 = ArrayListMultimap.create();
    private static final ListMultimap field_5844 = ArrayListMultimap.create();
    private static final Logger field_5845 = LogManager.getLogger();
@@ -104,9 +101,10 @@
 
    @KeepName
    public static void registerState(ModelIdentifier var0) {
-      if (var0.xrayVulnerable) {
-         field_5842.add(var0);
-      }
+//      if (var0.xrayVulnerable) {
+//         field_5842.add(var0);
+//      }
+      //FIXME: mixin fix
 
    }
 
@@ -324,7 +322,7 @@
       return null;
    }
 
-   private static void method_6147(ResourceManager var0, Identifier var1) {
+   private static void method_6147(ResourceManager var0, Identifier var1) throws IOException {
       String var2 = var1.getPath();
       byte var5 = -1;
       switch(var2.hashCode()) {
@@ -380,16 +378,18 @@
 
             while(var17.hasNext()) {
                Resource var6 = (Resource)var17.next();
-               int var7 = var6.getResourcePackPosition();
+               //FIXME: mixin fix
+               throw new RuntimeException("var 7 in Model Manager is not implemented");
+//               int var7 = var6.getResourcePackPosition();
 
-               while(((List)var3).size() <= var7) {
-                  ((List)var3).add((Object)null);
+//               while(((List)var3).size() <= var7) {
+//                  ((List)var3).add((Object)null);
+//               }
+//
+//               InputStreamReader var8 = new InputStreamReader(var6.getInputStream(), StandardCharsets.UTF_8);
+//               ((List)var3).set(var7, var8);
                }
-
-               InputStreamReader var8 = new InputStreamReader(var6.getInputStream(), StandardCharsets.UTF_8);
-               ((List)var3).set(var7, var8);
             }
-         }
 
          var14 = field_5844.get(var1);
          var14.addAll(Collections.nCopies(((List)var3).size(), (Object)null));
