--- a/net/badlion/class_1458.java
+++ b/net/badlion/class_1458.java
@@ -64,11 +64,16 @@
       this.field_3051 = new RectangleElement(-14869219);
       this.field_3051.relative(this.field_3078).setXAndY(0, 44).setWidthAndHeight(600, 633);
       class_1404 var4 = new class_1404(new class_2103() {
-         public void method_4013(class_1024.class_1025 var1) {
+         public void method_3851(class_1024.class_1025 var1) {
             class_1458.field_3058 = var1;
             class_1458.this.method_3990();
          }
       }, true, 10, 3, 32, 16, field_3074, "Organize") {
+         @Override
+         protected String method_3851(Object var1) {
+            return null;
+         }
+
          protected String method_4014(class_1024.class_1025 var1) {
             if (var1 == class_1024.class_1025.MOST_POPULAR) {
                return "Most Popular";
