--- a/net/badlion/class_1747.java
+++ b/net/badlion/class_1747.java
@@ -24,8 +24,8 @@
       this.field_4478 = new TextElement(-1, 18, class_1377.InputFlavor.field_2735);
       this.field_4479 = new TextElement(-1, 18, class_1377.InputFlavor.field_2735);
       this.field_4480 = new class_1404() {
-         protected String method_4839(String var1) {
-            return var1;
+         protected String method_3851(Object var1) {
+            return (String) var1;
          }
       };
       this.field_4480.field_2830.add("Linear");
