--- a/net/badlion/class_292.java
+++ b/net/badlion/class_292.java
@@ -32,10 +32,11 @@
    }
 
    public class_290 method_1111() {
-      class_290 var1 = this.method_1112();
-      var1.method_1284();
-      var1.method_1284();
-      return var1;
+      throw new RuntimeException("not implemented");
+//      class_290 var1 = this.method_1112();
+//      var1.method_1284();
+//      var1.method_1284();
+//      return var1;
    }
 
    public class_290 method_1112() {
@@ -47,7 +48,7 @@
       return var1;
    }
 
-   public void method_1392(JsonWriter var1, class_339 var2) {
+   public void method_1392(JsonWriter var1, class_339 var2) throws IOException {
       if (var2 instanceof class_340) {
          var1.value("linear");
       } else if (var2 instanceof class_337) {
@@ -65,7 +66,7 @@
 
    }
 
-   public class_339 method_1391(JsonReader var1) {
+   public class_339 method_1391(JsonReader var1) throws IOException {
       JsonToken var2 = var1.peek();
       if (var2 == JsonToken.STRING) {
          String var4 = var1.nextString();
@@ -85,7 +86,8 @@
 
    // $FF: synthetic method
    public class_348 method_1393() {
-      return this.method_1112();
+      throw new RuntimeException("not implemented");
+//      return this.method_1112();
    }
 
    static {
