--- a/net/badlion/class_228.java
+++ b/net/badlion/class_228.java
@@ -1,5 +1,12 @@
 package net.badlion;
 
 public interface class_228 {
-   Object method_906(Object var1, Object var2);
+    default Object method_906(Object var1, Object var2) {
+        throw new RuntimeException("not implemented");
 }
+
+    default class_214 method_885(class_214 var1, class_214 var2) {
+       throw new RuntimeException("not implemented");
+    }
+
+}
