--- a/net/badlion/class_865.java
+++ b/net/badlion/class_865.java
@@ -4,6 +4,8 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
+
+import io.github.badlionmoddinggroup.badlion.extra.EntityExtra;
 import net.badlion.client.common.entity.player.PlayerWrapper;
 import net.badlion.client.common.world.chunk.Chunk;
 import net.badlion.client.wrapper.MinecraftWrapper;
@@ -62,7 +64,7 @@
          var3 = (PlayerEntity)var2.next();
       } while(!var3.getUuid().equals(var1));
 
-      return var3.getBadlionEntity();
+      return (PlayerWrapper) ((EntityExtra)var3).getBadlionEntity();
    }
 
    public class_46 method_334(class_84 var1) {
@@ -80,7 +82,7 @@
 
       while(var15.hasNext()) {
          Entity var16 = (Entity)var15.next();
-         var14.add(var16.getBadlionEntity());
+         var14.add(((EntityExtra)var16).getBadlionEntity());
       }
 
       return var14;
@@ -108,7 +110,7 @@
 
       while(var2.hasNext()) {
          PlayerEntity var3 = (PlayerEntity)var2.next();
-         var1.add(var3.getBadlionEntity());
+         var1.add(((EntityExtra)var3).getBadlionEntity());
       }
 
       return var1;
@@ -120,7 +122,7 @@
 
       while(var2.hasNext()) {
          Entity var3 = (Entity)var2.next();
-         var1.add(var3.getBadlionEntity());
+         var1.add(((EntityExtra)var3).getBadlionEntity());
       }
 
       return var1;
