--- a/net/badlion/class_1013.java
+++ b/net/badlion/class_1013.java
@@ -7,15 +7,17 @@
 import org.apache.commons.lang3.mutable.MutableDouble;
 import org.apache.logging.log4j.LogManager;
 
+import java.io.IOException;
+
 public class class_1013 extends TypeAdapter {
-   public void method_3025(JsonWriter var1, MutableDouble var2) {
+   public void method_3025(JsonWriter var1, MutableDouble var2) throws IOException {
       var1.beginObject();
       var1.name("value");
       var1.value(var2.doubleValue());
       var1.endObject();
    }
 
-   public MutableDouble method_3024(JsonReader var1) {
+   public MutableDouble method_3024(JsonReader var1) throws IOException {
       MutableDouble var2 = new MutableDouble();
       String var3 = "";
       boolean var4 = false;
@@ -50,12 +52,12 @@
    }
 
    // $FF: synthetic method
-   public Object read(JsonReader var1) {
+   public Object read(JsonReader var1) throws IOException {
       return this.method_3024(var1);
    }
 
    // $FF: synthetic method
-   public void write(JsonWriter var1, Object var2) {
+   public void write(JsonWriter var1, Object var2) throws IOException {
       this.method_3025(var1, (MutableDouble)var2);
    }
 }
