--- a/net\badlion\Class_2057.java
+++ b/net\badlion\Class_2057.java
@@ -7,6 +7,8 @@
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonToken;
 import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.Locale;
 
@@ -26,7 +28,7 @@
             }
 
             return new TypeAdapter<T>() {
-                public void write(JsonWriter parameter_1, Object parameter_2) {
+                public void write(JsonWriter parameter_1, Object parameter_2) throws IOException {
                     if (parameter_2 == null) {
                         parameter_1.nullValue();
                     } else {
@@ -35,12 +37,12 @@
 
                 }
 
-                public T read(JsonReader parameter_1) {
+                public T read(JsonReader parameter_1) throws IOException {
                     if (parameter_1.peek() == JsonToken.NULL) {
                         parameter_1.nextNull();
                         return null;
                     } else {
-                        return var4.get(parameter_1.nextString());
+                        return (T) var4.get(parameter_1.nextString());
                     }
                 }
             };
