--- a/net/badlion/class_1188.java
+++ b/net/badlion/class_1188.java
@@ -9,10 +9,14 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Set;
+
+import io.github.badlionmoddinggroup.badlion.extra.EntityExtra;
 import net.badlion.client.common.BadlionClient;
+import net.badlion.client.common.entity.player.PlayerWrapper;
 import net.badlion.client.common.mod.SchematicaMod;
 import net.badlion.tags.KeepName;
 import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.class_1111;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.network.ClientPlayerEntity;
 import net.minecraft.client.render.BufferBuilder;
@@ -113,7 +117,8 @@
    private void method_3430() {
       this.field_2290 = new class_1142();
       this.field_2297 = new class_1144() {
-         public net.minecraft.class_1111 method_3261(World var1, WorldRenderer var2, BlockPos var3, int var4) {
+
+         public net.minecraft.class_1111 method_3831(World var1, WorldRenderer var2, BlockPos var3, int var4) {
             return new class_1148(var1, var2, var3, var4);
          }
 
@@ -126,7 +131,8 @@
    private void method_3432() {
       this.field_2290 = new class_1141();
       this.field_2297 = new class_1144() {
-         public net.minecraft.class_1111 method_3262(World var1, WorldRenderer var2, BlockPos var3, int var4) {
+
+         public net.minecraft.class_1111 method_3831(World var1, WorldRenderer var2, BlockPos var3, int var4) {
             return new class_1147(var1, var2, var3, var4);
          }
 
@@ -187,7 +193,7 @@
       ClientPlayerEntity var2 = this.field_2310.player;
       if (var2 != null) {
          this.field_2311.push("schematica");
-         SchematicaMod.method_2170(var2.getBadlionEntity(), var1);
+         SchematicaMod.method_2170((PlayerWrapper) ((EntityExtra) var2).getBadlionEntity(), var1);
          class_1159 var3 = (class_1159)SchematicaMod.field_1222;
          boolean var4 = var3 != null && var3.method_2644();
          SchematicaMod var5 = BadlionClient.getBadlion().getWrapper().getActiveModProfile().getSchematica();
@@ -414,7 +420,9 @@
 
          while(var12.hasNext()) {
             BlockEntity var13 = (BlockEntity)var12.next();
-            Box var14 = var13.getRenderBoundingBox();
+//            Box var14 = var13.getRenderBoundingBox();
+            //FIXME: yes
+            Box var14 = new Box(0,0,0,0,0,0);
             ++this.field_2294;
             if (var2.method_3845(var14)) {
                net.badlion.client.common.block.BlockPos var15 = this.field_2319.method_2649();
