--- a/net\badlion\Class_1013.java
+++ b/net\badlion\Class_1013.java
@@ -7,15 +7,17 @@
 import org.apache.commons.lang3.mutable.MutableDouble;
 import org.apache.logging.log4j.LogManager;
 
+import java.io.IOException;
+
 public class Class_1013 extends TypeAdapter<MutableDouble> {
-    public void method_3025(JsonWriter parameter_1, MutableDouble parameter_2) {
+    public void method_3025(JsonWriter parameter_1, MutableDouble parameter_2) throws IOException {
         parameter_1.beginObject();
         parameter_1.name("value");
         parameter_1.value(parameter_2.doubleValue());
         parameter_1.endObject();
     }
 
-    public MutableDouble method_3024(JsonReader parameter_1) {
+    public MutableDouble method_3024(JsonReader parameter_1) throws IOException {
         MutableDouble var2 = new MutableDouble();
         String var3 = "";
         boolean var4 = false;
@@ -48,4 +50,14 @@
         parameter_1.endObject();
         return var2;
     }
+
+    @Override
+    public void write(JsonWriter out, MutableDouble value) throws IOException {
+        method_3025(out, value);
 }
+
+    @Override
+    public MutableDouble read(JsonReader in) throws IOException {
+        return method_3024(in);
+    }
+}
