--- a/net/badlion/class_362.java
+++ b/net/badlion/class_362.java
@@ -2,6 +2,8 @@
 
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
 import java.io.StringReader;
 import java.io.StringWriter;
 import java.util.ArrayList;
@@ -9,7 +11,7 @@
 import java.util.List;
 
 public class class_362 {
-   public static List method_1334(String var0) {
+   public static List method_1334(String var0) throws IOException {
       ArrayList var1 = new ArrayList();
       JsonReader var2 = new JsonReader(new StringReader(var0));
       var2.beginArray();
@@ -22,7 +24,7 @@
       return var1;
    }
 
-   private static class_361 method_1333(JsonReader var0) {
+   private static class_361 method_1333(JsonReader var0) throws IOException {
       class_361 var1 = new class_361();
       var0.beginObject();
 
@@ -74,7 +76,7 @@
       }
    }
 
-   public static String method_1335(List var0) {
+   public static String method_1335(List var0) throws IOException {
       StringWriter var1 = new StringWriter();
       JsonWriter var2 = new JsonWriter(var1);
       var2.beginArray();
@@ -89,7 +91,7 @@
       return var1.toString();
    }
 
-   private static void method_1336(class_361 var0, JsonWriter var1) {
+   private static void method_1336(class_361 var0, JsonWriter var1) throws IOException {
       var1.beginObject();
       var1.name("realTimestamp").value((long)var0.method_1322());
       var1.name("value");
