--- a/net/badlion/class_1622.java
+++ b/net/badlion/class_1622.java
@@ -12,7 +12,7 @@
    private Map field_3752 = new HashMap();
    private List field_3753 = new LinkedList();
 
-   class_1622(class_578 var1, double var2, double var4) {
+   public class_1622(class_578 var1, double var2, double var4) {
       this.field_3749 = var1;
       this.field_3750 = var2;
       this.field_3751 = var4;
@@ -30,7 +30,7 @@
       return this.field_3751;
    }
 
-   protected void method_4418(int var1, int var2) {
+   public void method_4418(int var1, int var2) {
       this.field_3750 = -((double)var1 - this.field_3749.getX());
       this.field_3751 = -((double)var2 - this.field_3749.getY());
    }
