--- a/net\badlion\Class_365.java
+++ b/net\badlion\Class_365.java
@@ -1,6 +1,7 @@
 package net.badlion;
 
 import java.io.File;
+import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.text.SimpleDateFormat;
@@ -40,7 +41,7 @@
     private List<Class_361> field_543;
     private Class_2171 field_5987;
 
-    public Class_365(File parameter_1) {
+    public Class_365(File parameter_1) throws IOException {
         this.field_535 = parameter_1 == null;
         this.field_536 = new File(BadlionClient.minecraft.getRunDirectory(), "replay_recordings/");
         if (!this.field_536.exists()) {
@@ -135,7 +136,6 @@
             ReplayManager.field_5243 = true;
             ReplayManager.field_5242.submit(new Runnable() {
                 public void run() {
-                    try {
                         if (Class_365.this.field_541 != null) {
                             Class_365.this.field_539.method_1387(Class_365.this.field_541);
                         }
@@ -154,10 +154,11 @@
 
                         Class_365.this.method_1358();
                         if (!parameter_1) {
+                        try {
                             Class_365.this.method_1365();
+                        } catch (IOException e) {
+                            e.printStackTrace();
                         }
-                    } catch (IOException var2) {
-                        var2.printStackTrace();
                     }
 
                     ReplayManager.field_5243 = false;
@@ -200,7 +201,7 @@
 
     }
 
-    private void method_1365() {
+    private void method_1365() throws IOException {
         this.field_539 = new Class_370(this, this.field_535 ? this.field_537 : this.field_538, !this.field_535);
         if (!this.field_535) {
             this.field_540 = new Class_369(this.field_537);
@@ -208,8 +209,8 @@
 
     }
 
-    public Class_321 method_1364() {
-        return new Class_321(this.field_540.method_1368(), this.field_540.method_1374(), this.field_540.method_1373());
+    public Class_321 method_1364() throws IOException {
+        return new Class_321(this.field_540.method_1368(), this.field_540.method_1374(), (List<Class_361>) this.field_540.method_1373());
     }
 
     public Class_2171 method_6298() {
