--- a/net/badlion/class_819.java
+++ b/net/badlion/class_819.java
@@ -2,6 +2,8 @@
 
 import java.lang.reflect.Field;
 import java.util.List;
+
+import io.github.badlionmoddinggroup.badlion.util.BadlionKeyBinding;
 import net.badlion.client.common.keybind.KeybindWrapper;
 import net.minecraft.client.options.GameOptions;
 import net.minecraft.client.options.KeyBinding;
@@ -391,7 +393,7 @@
       for(int var3 = 0; var3 < var2.keysAll.length; ++var3) {
          KeyBinding var4 = var2.keysAll[var3];
          if (var4.getTranslationKey().equals(var1)) {
-            KeyBinding.onReplaceRemoveKeyBind(var4);
+            BadlionKeyBinding.onReplaceRemoveKeyBind(var4);
             class_1920 var5 = new class_1920(var4.getTranslationKey(), var4.getCode(), var4.getCategory());
             var2.keysAll[var3] = var5;
             if (var4.getTranslationKey().equals("of.key.zoom")) {
