--- a/net/badlion/class_1739.java
+++ b/net/badlion/class_1739.java
@@ -38,8 +38,8 @@
       this.field_4417.setLabel("Render", 16, AbstractFontRenderer.FontType.HEADER);
       this.field_4417.setBackground(-9648411, -7481611);
       this.field_4418 = new class_1404() {
-         protected String method_4826(String var1) {
-            return var1;
+         protected String method_3851(Object var1) {
+            return (String) var1;
          }
       };
       this.field_4418.field_2830.add("MP4 - Custom bitrate");
