--- a/net\badlion\Class_1156.java
+++ b/net\badlion\Class_1156.java
@@ -71,7 +71,7 @@
 
     }
 
-    public Class_780 method_3313(Class_705 parameter_1, Direction parameter_2, boolean parameter_3) {
+    public Class_780 method_3313(Class_705 parameter_1, Direction parameter_2, boolean parameter_3) throws Class_1157 {
         Vec3i var4 = this.method_3307(parameter_2, parameter_1.method_2185(), parameter_1.method_2187(), parameter_1.method_2186());
         Class_780 var5 = new Class_780(parameter_1.method_2183(), var4.getX(), var4.getY(), var4.getZ());
         BlockPos var6 = new BlockPos();
@@ -99,7 +99,7 @@
         return var5;
     }
 
-    private Vec3i method_3307(Direction parameter_1, int parameter_2, int parameter_3, int parameter_4) {
+    private Vec3i method_3307(Direction parameter_1, int parameter_2, int parameter_3, int parameter_4) throws Class_1157 {
         switch(parameter_1) {
         case DOWN:
         case UP:
@@ -115,7 +115,7 @@
         }
     }
 
-    private Class_84 method_3316(Class_84 parameter_1, Direction parameter_2, Vec3i parameter_3, BlockPos parameter_4) {
+    private Class_84 method_3316(Class_84 parameter_1, Direction parameter_2, Vec3i parameter_3, BlockPos parameter_4) throws Class_1157 {
         switch(parameter_2) {
         case DOWN:
             return parameter_4.method_273(parameter_1.getZ(), parameter_1.getY(), parameter_3.getZ() - 1 - parameter_1.getX());
@@ -134,7 +134,7 @@
         }
     }
 
-    private BlockState method_3306(BlockState parameter_1, Direction parameter_2, boolean parameter_3) {
+    private BlockState method_3306(BlockState parameter_1, Direction parameter_2, boolean parameter_3) throws Class_1157 {
         Property var4 = Class_1151.method_3297(parameter_1, "facing");
         if (var4 instanceof DirectionProperty) {
             Comparable var5 = parameter_1.get(var4);
