--- a/net/badlion/class_352.java
+++ b/net/badlion/class_352.java
@@ -2,6 +2,8 @@
 
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
 import java.util.Collection;
 
 public interface class_352 {
@@ -15,9 +17,9 @@
 
    Collection method_1289();
 
-   void method_1291(Object var1, Object var2);
+   default void method_1291(Object var1, Object var2) {}
 
-   void method_1290(JsonWriter var1, Object var2);
+   default void method_1290(JsonWriter var1, Object var2) {}
 
-   Object method_1293(JsonReader var1);
+   Object method_1293(JsonReader var1) throws IOException;
 }
