--- a/net\badlion\Class_440.java
+++ b/net\badlion\Class_440.java
@@ -1,10 +1,11 @@
 package net.badlion;
 
+import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 
 public class Class_440 {
-    public static int method_1595(InputStream parameter_0) {
+    public static int method_1595(InputStream parameter_0) throws IOException {
         int var1 = parameter_0.read();
         int var2 = parameter_0.read();
         int var3 = parameter_0.read();
@@ -12,7 +13,7 @@
         return (var1 | var2 | var3 | var4) < 0 ? -1 : var1 << 24 | var2 << 16 | var3 << 8 | var4;
     }
 
-    public static void method_1597(OutputStream parameter_0, int parameter_1) {
+    public static void method_1597(OutputStream parameter_0, int parameter_1) throws IOException {
         parameter_0.write(parameter_1 >>> 24 & 255);
         parameter_0.write(parameter_1 >>> 16 & 255);
         parameter_0.write(parameter_1 >>> 8 & 255);
@@ -49,19 +50,19 @@
                 this.field_747 = true;
             }
 
-            public int read() {
+            public int read() throws IOException {
                 return this.field_747 ? -1 : parameter_0.read();
             }
 
-            public int read(byte[] parameter_1, int parameter_2, int parameter_3) {
+            public int read(byte[] parameter_1, int parameter_2, int parameter_3) throws IOException {
                 return this.field_747 ? -1 : parameter_0.read(parameter_1, parameter_2, parameter_3);
             }
 
-            public int available() {
+            public int available() throws IOException {
                 return parameter_0.available();
             }
 
-            public long skip(long parameter_1) {
+            public long skip(long parameter_1) throws IOException {
                 return this.field_747 ? 0L : parameter_0.skip(parameter_1);
             }
 
@@ -69,7 +70,7 @@
                 parameter_0.mark(parameter_1);
             }
 
-            public synchronized void reset() {
+            public synchronized void reset() throws IOException {
                 parameter_0.reset();
             }
 
@@ -77,13 +78,13 @@
                 return parameter_0.markSupported();
             }
 
-            public int read(byte[] parameter_1) {
+            public int read(byte[] parameter_1) throws IOException {
                 return this.field_747 ? -1 : parameter_0.read(parameter_1);
             }
         };
     }
 
-    public static void method_1596(InputStream parameter_0, OutputStream parameter_1) {
+    public static void method_1596(InputStream parameter_0, OutputStream parameter_1) throws IOException {
         byte[] var2 = new byte[1024];
 
         int var3;
