--- a/net/badlion/class_1009.java
+++ b/net/badlion/class_1009.java
@@ -6,14 +6,16 @@
 import net.badlion.client.common.BadlionClient;
 import net.badlion.client.common.input.Keyboard;
 
+import java.io.IOException;
+
 public class class_1009 extends TypeAdapter {
-   public void method_3019(JsonWriter var1, Keyboard.Keybind var2) {
+   public void method_3019(JsonWriter var1, Keyboard.Keybind var2) throws IOException {
       var1.value(var2.name());
    }
 
-   public Keyboard.Keybind method_3018(JsonReader var1) {
+   public Keyboard.Keybind method_3018(JsonReader var1) throws IOException {
       try {
-         int var2;
+         int var2 = 0;
          try {
             var1.beginObject();
             var1.nextName();
@@ -21,6 +23,8 @@
             var1.endObject();
          } catch (IllegalStateException var6) {
             var2 = var1.nextInt();
+         } catch (IOException e) {
+            e.printStackTrace();
          }
 
          String var3 = BadlionClient.getBadlion().getWrapper().getModProfileManager().method_378();
@@ -61,18 +65,18 @@
          } else {
             return Keyboard.Keybind.KEY_UNKNOWN;
          }
-      } catch (NumberFormatException var7) {
+      } catch (NumberFormatException | IOException var7) {
          return Keyboard.Keybind.valueOf(var1.nextString());
       }
    }
 
    // $FF: synthetic method
-   public Object read(JsonReader var1) {
+   public Object read(JsonReader var1) throws IOException {
       return this.method_3018(var1);
    }
 
    // $FF: synthetic method
-   public void write(JsonWriter var1, Object var2) {
+   public void write(JsonWriter var1, Object var2) throws IOException {
       this.method_3019(var1, (Keyboard.Keybind)var2);
    }
 }
