--- a/net/badlion/class_354.java
+++ b/net/badlion/class_354.java
@@ -7,7 +7,13 @@
 
    double method_1305();
 
-   double method_1302(Object var1);
+    default double method_1302(Object var1) {
+       throw new RuntimeException("not impl");
+    }
 
-   Object method_1303(Object var1, double var2);
+
+
+    default Object method_1303(Object var1, double var2) {
+       throw new RuntimeException("not impl");
 }
+}
