--- a/net/badlion/class_838.java
+++ b/net/badlion/class_838.java
@@ -22,7 +22,7 @@
 
    public void method_2591(SessionWrapper var1, final class_881.class_882 var2, boolean var3) {
       this.field_2127.loadSkin(((Session)var1.method_22()).getProfile(), new SkinTextureAvailableCallback() {
-         public void method_3170(Type var1, Identifier var2x, MinecraftProfileTexture var3) {
+         public void method_4402(Type var1, Identifier var2x, MinecraftProfileTexture var3) {
             var2.method_2592(class_881.class_883.values()[var1.ordinal()], new MinecraftTexture(var2x), class_838.this.method_2482(var3));
          }
       }, var3);
