--- a/net/badlion/class_1926.java
+++ b/net/badlion/class_1926.java
@@ -123,14 +123,14 @@
 
                switch(var14) {
                case 0:
-                  if (var2 == Mode.field_4646) {
-                     var8 = var10 ? (var11 ? "7_opti_fishing_rod" : "7_fishing_rod") : var8;
-                  }
+//                  if (var2 == Mode.field_4646) {//field_4646
+//                     var8 = var10 ? (var11 ? "7_opti_fishing_rod" : "7_fishing_rod") : var8;
+//                  }
                   break;
                case 1:
-                  if (var2 == Mode.field_4646) {
-                     var8 = var10 ? (var11 ? "7_opti_fishing_rod_cast" : "7_fishing_rod_cast") : var8;
-                  }
+//                  if (var2 == Mode.field_4646) {//field_4646
+//                     var8 = var10 ? (var11 ? "7_opti_fishing_rod_cast" : "7_fishing_rod_cast") : var8;
+//                  }
                   break;
                case 2:
                   var8 = var15 ? (var11 ? "7_opti_bow_pulling_2" : "7_bow_pulling_2") : var8;
@@ -150,36 +150,37 @@
 
             var15 = Wrapper.getWrapper().getActiveModProfile().getAnimations().isEnabled() && Wrapper.getWrapper().getActiveModProfile().getAnimations().method_1694().isTrue();
             var10 = Wrapper.getWrapper().getActiveModProfile().getAnimations().isEnabled() && Wrapper.getWrapper().getActiveModProfile().getAnimations().method_1699().isTrue();
-            if (var2 == Mode.field_4646) {
-               if (var15 && var0.getUseAction() == UseAction.BLOCK && ((PlayerEntity)var1).method_8009() > 0) {
-                  BadlionClient.glStateManager.method_89(0.0F, 0.0F, 0.0035F);
+//            if (var2 == Mode.field_4646) {
+//               if (var15 && var0.getUseAction() == UseAction.BLOCK && ((PlayerEntity)var1).method_8009() > 0) {
+//                  BadlionClient.glStateManager.method_89(0.0F, 0.0F, 0.0035F);
+//               }
+//
+//               if (var0.getItem() == Items.FISHING_ROD && Wrapper.getWrapper().getActiveModProfile().getAnimations().isEnabled() && Wrapper.getWrapper().getActiveModProfile().getAnimations().method_1705().isTrue()) {
+//                  GlStateManager.rotatef(180.0F, 0.0F, 1.0F, 0.0F);
+//               }
+//            } else if (var2 == Mode.field_4645) {
+//               if (var15 && var0.getUseAction() == UseAction.BLOCK && ((PlayerEntity)var1).method_8009() > 0) {
+//                  BadlionClient.glStateManager.method_84(-35.0F, 0.0F, 1.0F, 0.0F);
+//                  BadlionClient.glStateManager.method_84(-15.0F, 1.0F, 0.0F, 0.0F);
+//                  BadlionClient.glStateManager.method_84(-60.0F, 0.0F, 0.0F, 1.0F);
+//               } else if (var10) {
+//                  if (var6.isHandheld()) {
+//                     BadlionClient.glStateManager.method_89(-0.01F, 0.0F, 0.098F);
+//                     BadlionClient.glStateManager.method_84(-15.0F, 1.0F, 0.0F, 0.0F);
+//                     BadlionClient.glStateManager.method_84(5.0F, 0.0F, 0.0F, 1.0F);
+//                  } else {
+//                     BadlionClient.glStateManager.method_81(0.0D, 0.03999999910593033D, 0.0D);
+//                     BadlionClient.glStateManager.method_84(-14.0F, 1.0F, 0.0F, 0.0F);
+//                     BadlionClient.glStateManager.method_84(8.0F, 0.0F, 0.0F, 1.0F);
+//                     BadlionClient.glStateManager.method_84(-14.0F, 0.0F, 1.0F, 0.0F);
+//                  }
+//               }
                }
+//         }
 
-               if (var0.getItem() == Items.FISHING_ROD && Wrapper.getWrapper().getActiveModProfile().getAnimations().isEnabled() && Wrapper.getWrapper().getActiveModProfile().getAnimations().method_1705().isTrue()) {
-                  GlStateManager.rotatef(180.0F, 0.0F, 1.0F, 0.0F);
+//         var3.method_3983(var0, var4, var2);
                }
-            } else if (var2 == Mode.field_4645) {
-               if (var15 && var0.getUseAction() == UseAction.BLOCK && ((PlayerEntity)var1).method_8009() > 0) {
-                  BadlionClient.glStateManager.method_84(-35.0F, 0.0F, 1.0F, 0.0F);
-                  BadlionClient.glStateManager.method_84(-15.0F, 1.0F, 0.0F, 0.0F);
-                  BadlionClient.glStateManager.method_84(-60.0F, 0.0F, 0.0F, 1.0F);
-               } else if (var10) {
-                  if (var6.isHandheld()) {
-                     BadlionClient.glStateManager.method_89(-0.01F, 0.0F, 0.098F);
-                     BadlionClient.glStateManager.method_84(-15.0F, 1.0F, 0.0F, 0.0F);
-                     BadlionClient.glStateManager.method_84(5.0F, 0.0F, 0.0F, 1.0F);
-                  } else {
-                     BadlionClient.glStateManager.method_81(0.0D, 0.03999999910593033D, 0.0D);
-                     BadlionClient.glStateManager.method_84(-14.0F, 1.0F, 0.0F, 0.0F);
-                     BadlionClient.glStateManager.method_84(8.0F, 0.0F, 0.0F, 1.0F);
-                     BadlionClient.glStateManager.method_84(-14.0F, 0.0F, 1.0F, 0.0F);
-                  }
-               }
-            }
-         }
 
-         var3.method_3983(var0, var4, var2);
-      }
 
    }
 
