--- a/net/badlion/class_812.java
+++ b/net/badlion/class_812.java
@@ -14,23 +14,26 @@
         return this.field_2109;
     }
 
+   //TODO: MIXIN into title screen and fix
+
     public void method_173() {
-        this.field_2109.getGuiPatchNotes().method_4127(true);
+//      this.field_2109.getGuiPatchNotes().method_4127(true);
     }
 
     public void method_174() {
-        this.field_2109.getGuiCosmeticsMenu().open();
+//      this.field_2109.getGuiCosmeticsMenu().open();
     }
 
     public void method_175() {
-        this.field_2109.getGuiCosmeticsMenu().method_4094();
+//      this.field_2109.getGuiCosmeticsMenu().method_4094();
     }
 
     public void method_177() {
-        this.field_2109.getGuiBadlionMainMenu().method_3953().open();
+//      this.field_2109.getGuiBadlionMainMenu().method_3953().open();
     }
 
     public boolean method_176() {
-        return this.field_2109.getGuiCosmeticsMenu().method_4713();
+//      return this.field_2109.getGuiCosmeticsMenu().method_4713();
+      return true;
     }
 }
