--- a/net/badlion/class_1120.java
+++ b/net/badlion/class_1120.java
@@ -1,5 +1,6 @@
 package net.badlion;
 
+import io.github.badlionmoddinggroup.badlion.extra.MinecraftClientExtra;
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.embedded.EmbeddedChannel;
 import net.badlion.client.common.manager.replay.ReplayManager;
@@ -33,9 +34,9 @@
     public void method_5336() {
         LogManager.getLogger().info("Start world called: open: " + this.field_5265);
         if (!this.field_5265) {
-            this.field_2160 = new ClientConnection(NetworkSide.field_6044);
+         this.field_2160 = new ClientConnection(NetworkSide.CLIENTBOUND);
             this.field_2160.setPacketListener(new ClientLoginNetworkHandler(this.field_2160, MinecraftClient.getInstance(), (Screen)null));
-            this.field_2161 = new EmbeddedChannel(new ChannelHandler[]{this.field_2160});
+         this.field_2161 = new EmbeddedChannel(this.field_2160);
             this.field_2161.pipeline().fireChannelActive();
             this.field_5265 = true;
             this.field_5266 = false;
@@ -63,9 +64,9 @@
         LogManager.getLogger().info("restart world called: open: " + this.field_5265);
         this.field_2161.close();
         MinecraftClient.getInstance().method_2310();
-        MinecraftClient.getInstance().disconnect((ClientWorld)null);
+      MinecraftClient.getInstance().disconnect(null);
         MinecraftClient.getInstance().openScreen(new class_1256());
-        MinecraftClient.getInstance().clearScheduledTasks();
+      ((MinecraftClientExtra)MinecraftClient.getInstance()).clearScheduledTasks();
         this.field_5265 = false;
         this.method_5336();
     }
@@ -102,6 +103,6 @@
     }
 
     public void method_5337(float var1) {
-        MinecraftClient.getInstance().getTimer().field_2799 = var1;
+      ((MinecraftClientExtra)MinecraftClient.getInstance()).getTimer().field_2799 = var1;
     }
 }
