--- a/net/badlion/class_370.java
+++ b/net/badlion/class_370.java
@@ -4,11 +4,7 @@
 import com.google.gson.JsonObject;
 import com.google.gson.JsonPrimitive;
 import java.awt.image.BufferedImage;
-import java.io.BufferedOutputStream;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.OutputStream;
+import java.io.*;
 import java.nio.charset.StandardCharsets;
 import java.util.Collections;
 import java.util.HashSet;
@@ -40,7 +36,7 @@
    private Set field_564;
    private class_2171 field_5988;
 
-   public class_370(class_365 var1, File var2, boolean var3) {
+   public class_370(class_365 var1, File var2, boolean var3) throws FileNotFoundException {
       this(new BufferedOutputStream(new FileOutputStream(var2)), var1, var2, var3);
    }
 
@@ -219,14 +215,11 @@
       }
    }
 
-   public void method_1388(boolean var1) {
+   public void method_1388(boolean var1) throws IOException {
       this.field_556 = var1;
       if (var1) {
-         try {
             this.putNextEntry(new ZipEntry("recording.tmcpr"));
             this.field_557.add("recording.tmcpr");
-         } catch (IOException var4) {
-         }
       } else {
          if (this.field_558 != null) {
             this.method_1387(this.field_558);
@@ -268,7 +261,7 @@
 
    }
 
-   public class_370.class_371 method_1380() {
+   public class_370.class_371 method_1380() throws IOException {
       boolean var1 = false;
       if (this.field_555) {
          this.field_555 = false;
