--- a/net/badlion/class_813.java
+++ b/net/badlion/class_813.java
@@ -2,6 +2,8 @@
 
 import java.util.HashMap;
 import java.util.Map;
+
+import io.github.badlionmoddinggroup.badlion.extra.ContainerExtra;
 import net.badlion.client.common.gui.lib.GuiType;
 import net.minecraft.client.gui.TwitchErrorScreen;
 import net.minecraft.client.gui.screen.AchievementsScreen;
@@ -81,7 +83,8 @@
    }
 
    public boolean isButtonOccludedByOtherGui(int var1, int var2) {
-      return this.field_2110.isButtonOccludedByOtherGui(var1, var2);
+//      return this.field_2110.isButtonOccludedByOtherGui(var1, var2);
+      return true;
    }
 
    public boolean method_5259() {
@@ -108,7 +111,7 @@
    }
 
    public class_565 method_5263() {
-      return this.field_2110 instanceof ChestScreen ? ((ChestScreen)this.field_2110).container.getBadlionContainer() : null;
+      return this.field_2110 instanceof ChestScreen ? ((ContainerExtra)((ChestScreen)this.field_2110).container).getBadlionContainer() : null;
    }
 
    public boolean method_5264() {
