--- a/net/badlion/class_778.java
+++ b/net/badlion/class_778.java
@@ -2,6 +2,8 @@
 
 import java.io.File;
 import java.io.FileInputStream;
+import java.io.IOException;
+
 import net.badlion.client.common.BadlionClient;
 import net.badlion.client.common.block.Blocks;
 import net.badlion.client.common.item.ItemStackWrapper;
@@ -13,7 +15,7 @@
 public class class_778 {
    public static final ItemStackWrapper field_1429;
 
-   public static ITag method_2368(File var0) {
+   public static ITag method_2368(File var0) throws IOException {
       try {
          return BadlionClient.minecraft.method_195().method_26(new FileInputStream(var0));
       } catch (Exception var2) {
