--- a/net\badlion\Class_1258.java
+++ b/net\badlion\Class_1258.java
@@ -1,5 +1,6 @@
 package net.badlion;
 
+import java.io.IOException;
 import java.io.InputStream;
 import java.nio.ByteBuffer;
 import net.minecraft.client.resource.metadata.TextureResourceMetadata;
@@ -26,7 +27,7 @@
         this.field_2415 = parameter_1;
     }
 
-    public void method_3456(ResourceManager parameter_1) {
+    public void method_3456(ResourceManager parameter_1) throws IOException {
         InputStream var2 = null;
 
         try {
@@ -54,6 +55,8 @@
             }
 
             this.field_2412 = true;
+        } catch (IOException e) {
+            e.printStackTrace();
         } finally {
             if (var2 != null) {
                 var2.close();
