--- a/net/badlion/class_1888.java
+++ b/net/badlion/class_1888.java
@@ -3,6 +3,9 @@
 import java.io.IOException;
 import java.util.Iterator;
 import java.util.LinkedList;
+
+import io.github.badlionmoddinggroup.badlion.gui.BadlionButtonWidget;
+import io.github.badlionmoddinggroup.badlion.gui.ButtonType;
 import net.badlion.client.common.BadlionClient;
 import net.badlion.client.common.mods.IMod;
 import net.badlion.client.common.text.AbstractFontRenderer;
@@ -94,11 +97,7 @@
          this.field_5123 = false;
       }
 
-      try {
          super.mouseClicked(var1, var2, var3);
-      } catch (IOException var8) {
-         var8.printStackTrace();
-      }
 
    }
 
@@ -122,13 +121,13 @@
       class_2093 var1 = new class_2093();
       byte var2 = 120;
       byte var3 = 50;
-      this.buttons.add(new ButtonWidget(0, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var3, var2, 20, "Back", .avs$ButtonType.THICK_LINES));
+      this.buttons.add(new BadlionButtonWidget(0, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var3, var2, 20, "Back", ButtonType.THICK_LINES));
       int var4 = var3 + 24;
-      this.buttons.add(new ButtonWidget(1, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var4, var2, 20, "Reset defaults", .avs$ButtonType.THICK_LINES));
+      this.buttons.add(new BadlionButtonWidget(1, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var4, var2, 20, "Reset defaults", ButtonType.THICK_LINES));
       var4 += 24;
-      this.buttons.add(new ButtonWidget(2, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var4, var2, 20, "Help", .avs$ButtonType.THICK_LINES));
+      this.buttons.add(new BadlionButtonWidget(2, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var4, var2, 20, "Help", ButtonType.THICK_LINES));
       var4 += 24;
-      this.buttons.add(3, new ButtonWidget(3, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var4, var2, 20, this.method_5125(), .avs$ButtonType.THICK_LINES));
+      this.buttons.add(3, new BadlionButtonWidget(3, var1.method_5687() / 2 - var2 / 2, var1.method_5688() - var4, var2, 20, this.method_5125(), ButtonType.THICK_LINES));
    }
 
    private String method_5125() {
