--- a/net/badlion/class_1927.java
+++ b/net/badlion/class_1927.java
@@ -34,11 +34,14 @@
 import org.lwjgl.opengl.GL11;
 
 @class_907
+//         FIXME: fix mixin
+// WHOLE CLASS
 public class class_1927 {
    public static void resizeFramebuffer1(MinecraftClient var0) {
       class_432 var1 = BadlionClient.getBadlion().getWrapper().getActiveModProfile().getReplay().method_1187();
       if (var1.method_1544() && ReplayOverlay.needsFrameBuffer) {
-         var0.privateResize(var1.method_1553(), var1.method_1554());
+//                  FIXME: fix mixin
+//         var0.privateResize(var1.method_1553(), var1.method_1554());
       }
 
    }
@@ -52,7 +55,8 @@
       }
 
       if (var1.method_1544() && ReplayOverlay.needsFrameBuffer) {
-         var0.privateResize(var1.method_1550(), var1.method_1552());
+//         FIXME: fix mixin
+//         var0.privateResize(var1.method_1550(), var1.method_1552());
          GL11.glViewport(0, 0, var1.method_1550(), var1.method_1552());
       }
 
@@ -81,11 +85,11 @@
 
                while(var2.hasNext()) {
                   var3 = (Entity)var2.next();
-                  if (var3 instanceof OtherClientPlayerEntity && class_505.isNearNPC(var3.getBadlionEntity())) {
-                     var2.remove();
+//                  if (var3 instanceof OtherClientPlayerEntity && class_505.isNearNPC(var3.getBadlionEntity())) {
+//                     var2.remove();
+//                  }
                   }
                }
-            }
 
             if (var1.ignoreItemFrameClicks.isTrue() && !class_2066.method_5571()) {
                var2 = var0.iterator();
@@ -110,9 +114,9 @@
                return false;
             }
 
-            if (var9.hidePlayersNearNPCs.isTrue() && var1 instanceof OtherClientPlayerEntity && class_505.isNearNPC(var1.getBadlionEntity())) {
-               return false;
-            }
+//            if (var9.hidePlayersNearNPCs.isTrue() && var1 instanceof OtherClientPlayerEntity && class_505.isNearNPC(var1.getBadlionEntity())) {
+//               return false;
+//            }
 
             if (var9.hidePlayersInLobby.isTrue()) {
                class_502 var10 = class_502.getCurrentLocation();
