--- a/net/badlion/class_372.java
+++ b/net/badlion/class_372.java
@@ -3,10 +3,12 @@
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
 
+import java.io.IOException;
+
 public interface class_372 {
     class_348 method_1393();
 
-    void method_1392(JsonWriter var1, class_339 var2);
+   void method_1392(JsonWriter var1, class_339 var2) throws IOException;
 
-    class_339 method_1391(JsonReader var1);
+   class_339 method_1391(JsonReader var1) throws IOException;
 }
