--- a/net/badlion/class_831.java
+++ b/net/badlion/class_831.java
@@ -1,5 +1,6 @@
 package net.badlion;
 
+import io.github.badlionmoddinggroup.badlion.extra.EntityExtra;
 import net.badlion.client.common.entity.EntityWrapper;
 import net.minecraft.util.hit.HitResult;
 import net.minecraft.util.math.Vec3d;
@@ -20,7 +21,7 @@
     }
 
     public EntityWrapper method_547() {
-        return this.field_2143.entitiy == null ? null : this.field_2143.entitiy.getBadlionEntity();
+      return this.field_2143.entitiy == null ? null : ((EntityExtra)this.field_2143.entitiy).getBadlionEntity();
     }
 
     public class_43 method_548() {
