--- a/net/badlion/class_850.java
+++ b/net/badlion/class_850.java
@@ -31,15 +31,15 @@
       this.field_2137.loadShader(((MinecraftTexture)var1).getResourceLocation());
    }
 
-   public List method_57() {
-      LinkedList var1 = new LinkedList();
+   public List<class_826> method_57() {
+      LinkedList<class_826> var1 = new LinkedList();
       if (this.field_2137.getShader() == null) {
          return var1;
       } else {
-         Iterator var2 = this.field_2137.getShader().getListShaders().iterator();
+         Iterator<PostProcessShader> var2 = this.field_2137.getShader().passes.iterator();
 
          while(var2.hasNext()) {
-            PostProcessShader var3 = (PostProcessShader)var2.next();
+            PostProcessShader var3 = var2.next();
             var1.add(new class_826(var3));
          }
 
