--- a/net/badlion/class_766.java
+++ b/net/badlion/class_766.java
@@ -1,5 +1,6 @@
 package net.badlion;
 
+import io.github.badlionmoddinggroup.badlion.extra.ItemStackExtra;
 import net.badlion.client.common.item.ItemStackWrapper;
 import net.minecraft.container.BeaconContainer;
 import net.minecraft.container.ChestInventory;
@@ -50,7 +51,7 @@
       if (var2 != null) {
          ItemStack var3 = var2.getStack();
          if (var3 != null) {
-            return var3.getBadlionItemStack();
+            return ((ItemStackExtra)(Object)var3).getBadlionItemStack();
          }
       }
 
