--- a/net/badlion/class_2143.java
+++ b/net/badlion/class_2143.java
@@ -74,7 +74,7 @@
    }
 
    public ModelTransformation method_6143() {
-      return new ModelTransformation(this.method_6136(Mode.field_4645), this.method_6136(Mode.field_4646), this.method_6136(Mode.field_4647), this.method_6136(Mode.field_4648), this.method_6136(Mode.field_4649), this.method_6136(Mode.field_4650));
+      return new ModelTransformation(this.method_6136(Mode.THIRD_PERSON), this.method_6136(Mode.FIRST_PERSON), this.method_6136(Mode.HEAD), this.method_6136(Mode.GUI), this.method_6136(Mode.GROUND), this.method_6136(Mode.FIXED));
    }
 
    private Transformation method_6136(Mode var1) {
