--- a/net/badlion/client/common/gui/lib/button/IconButton.java
+++ b/net/badlion/client/common/gui/lib/button/IconButton.java
@@ -9,7 +9,7 @@
    public TextureWrapper field_3209;
    public int field_3210 = -1;
    public int field_3211 = -1;
-   boolean field_3212;
+   public boolean field_3212;
    private boolean field_3213;
 
    public IconButton(Button.ButtonListener var1, TextureWrapper var2, TextureWrapper var3) {
