--- a/net\badlion\client\common\manager\CosmeticManager.java
+++ b/net\badlion\client\common\manager\CosmeticManager.java
@@ -3,6 +3,8 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
+
+import io.github.badlionmoddinggroup.badlion.extra.PlayerEntityExtra;
 import net.badlion.Class_1111;
 import net.badlion.Class_1112;
 import net.badlion.Class_14;
@@ -130,7 +132,7 @@
         boolean var7 = parameter_0.method_7177(2) != null;
         boolean var8 = parameter_0.method_7177(3) != null;
         boolean var9 = parameter_0.getModel().equals("slim");
-        return new Class_966(var2, var3.getPath(), var4, var5, var6, var7, var8, var9, parameter_1, false, parameter_0.controller.isAnimationPlaying());
+        return new Class_966(var2, var3.getPath(), var4, var5, var6, var7, var8, var9, parameter_1, false, ((PlayerEntityExtra)parameter_0).getController().isAnimationPlaying());
     }
 
     public static void method_2653(Class_966 parameter_0, AbstractClientPlayerEntity parameter_1, float parameter_2) {
