--- a/net\badlion\client\common\gui\overlay\ChatOverlay.java
+++ b/net\badlion\client\common\gui\overlay\ChatOverlay.java
@@ -1200,7 +1200,7 @@
                             }
                         }
                     } else if (Class_2133.method_5894(parameter_1, parameter_2, 0, this.field_4270, field_4223, this.method_4780() + 45)) {
-                        final Class_2086 var21;
+                        Class_2086 var21;
                         if (var5.getSelectedChat() != null) {
                             var21 = var5.getSelectedChat().method_4572(this.field_2764, this.field_2763, (double)this.field_2761, this.field_4270, this.method_4780(), parameter_2);
                             Class_1679 var24 = (Class_1679)var21.method_5666();
@@ -1233,14 +1233,15 @@
                                     switch((Integer)var21.method_5670()) {
                                     case 0:
                                     case 1:
+                                        Class_2086 finalVar2 = var21;
                                         BadlionClient.chatManager.method_4769().execute(new Runnable() {
                                             public void run() {
                                                 String var1 = "";
 
                                                 try {
-                                                    Class_946 var2 = new Class_946(UUID.fromString(var26[0]), (Integer)var21.method_5670() == 0, "Muted via BLC", 86400000L, var26[2]);
+                                                    Class_946 var2 = new Class_946(UUID.fromString(var26[0]), (Integer) finalVar2.method_5670() == 0, "Muted via BLC", 86400000L, var26[2]);
                                                     if (!FriendsOverlay.method_4678(var1 = BadlionClient.getBadlion().getWrapper().chatMuteUser(BadlionClient.UGLY_GSON.toJson(var2)), "Chat")) {
-                                                        BadlionClient.getBadlion().getWrapper().getModProfile().addNotification(new Class_700(5000L, "Chat", Class_699.SUCCESS, "You've " + ((Integer)var21.method_5670() == 0 ? "muted" : "unmuted") + " " + var26[1]));
+                                                        BadlionClient.getBadlion().getWrapper().getModProfile().addNotification(new Class_700(5000L, "Chat", Class_699.SUCCESS, "You've " + ((Integer) finalVar2.method_5670() == 0 ? "muted" : "unmuted") + " " + var26[1]));
                                                         LogManager.getLogger().debug("Notification send");
                                                     }
                                                 } catch (Exception var3) {
