--- a/net\badlion\client\common\gui\overlay\EmoteWheelSubclassOverlay.java
+++ b/net\badlion\client\common\gui\overlay\EmoteWheelSubclassOverlay.java
@@ -1,5 +1,6 @@
 package net.badlion.client.common.gui.overlay;
 
+import io.github.badlionmoddinggroup.badlion.extra.PlayerEntityExtra;
 import net.badlion.Class_1042;
 import net.badlion.Class_1306;
 import net.badlion.Class_2027;
@@ -18,14 +19,14 @@
             if (var4.field_2567.contains(parameter_1)) {
                 int var5 = BadlionClient.getBadlion().getModelManager().method_3535(parameter_1);
                 Class_2027 var6 = BadlionClient.getBadlion().getModelManager().method_3531(var5);
-                var3.controller.method_3256(var3, var6);
+                ((PlayerEntityExtra)var3).getController().method_3256(var3, var6);
                 Class_1306.method_3530(var5);
                 if (BadlionClient.getBadlion().getReplayManager().method_5309()) {
                     BadlionClient.getBadlion().getReplayManager().method_6274().method_6304(new Class_1042(var3.getUuid(), CosmeticType.EMOTE, var5));
                 }
             }
         } else {
-            var3.controller.currentAnimation = null;
+            ((PlayerEntityExtra)var3).getController().currentAnimation = null;
         }
 
     }
