--- a/net/badlion/client/common/gui/overlay/FriendsSettingsOverlay.java
+++ b/net/badlion/client/common/gui/overlay/FriendsSettingsOverlay.java
@@ -59,8 +59,38 @@
             }
 
          }
+
+         @Override
+         public void method_6186(MutableBoolean var1) {
+
+         }
+
+         @Override
+         public void method_4846(class_1363 var1) {
+
+         }
+
+         @Override
+         public void method_4845(class_1690 var1) {
+
+         }
       });
       this.field_4501 = new class_1844("", class_1690.values()[this.field_4493.field_4146], FriendsOverlay.field_4120, new class_2103() {
+         @Override
+         public void method_4844(class_1691 var1) {
+
+         }
+
+         @Override
+         public void method_6186(MutableBoolean var1) {
+
+         }
+
+         @Override
+         public void method_4846(class_1363 var1) {
+
+         }
+
          public void method_4845(class_1690 var1) {
             int var2 = FriendsSettingsOverlay.this.field_4493.field_4146;
             int var3 = FriendsSettingsOverlay.this.field_4493.field_4146 = var1.ordinal();
@@ -79,17 +109,47 @@
       });
       final FriendsMenu var2 = BadlionClient.getBadlion().getWrapper().getActiveModProfile().getFriendSettings();
       this.field_5929 = new ModFeatureEnabledBool("Show Current Server to Friends", var2.method_6174(), new class_2103() {
+         @Override
+         public void method_4844(class_1691 var1) {
+
+         }
+
          public void method_6186(MutableBoolean var1) {
             class_1045 var2x = new class_1045(FriendsSettingsOverlay.this.field_4493.field_4146, FriendsSettingsOverlay.this.field_4493.field_4145, var2.method_6174().booleanValue());
             String var3 = BadlionClient.getBadlion().getWrapper().friendsSettingsUpdate(BadlionClient.UGLY_GSON.toJson(var2x));
             FriendsOverlay.method_4677(var3);
          }
+
+         @Override
+         public void method_4846(class_1363 var1) {
+
+         }
+
+         @Override
+         public void method_4845(class_1690 var1) {
+
+         }
       });
       this.field_5930 = new ModFeatureEnabledBool("Notify Friend Status Change", var2.method_6175(), "Show a notification when your friend goes online or offline");
       this.field_4502 = new class_1844("", var2.method_3650(), class_1363.values(), new class_2103() {
+         @Override
+         public void method_4844(class_1691 var1) {
+
+         }
+
+         @Override
+         public void method_6186(MutableBoolean var1) {
+
+         }
+
          public void method_4846(class_1363 var1) {
             var2.method_3647(var1);
          }
+
+         @Override
+         public void method_4845(class_1690 var1) {
+
+         }
       });
       this.field_4503 = new class_1472("Toggle Friends keybind:", FriendsMenu.OPEN_KEY);
       this.field_4504 = new class_1855("", var2.method_3648(), 0.0D, 1.0D, 0.01D);
