--- a/net/badlion/client/Wrapper.java
+++ b/net/badlion/client/Wrapper.java
@@ -102,6 +102,7 @@
 import org.apache.logging.log4j.Level;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.core.Logger;
+import org.lwjgl.LWJGLException;
 import org.lwjgl.opengl.ContextCapabilities;
 import org.lwjgl.opengl.Display;
 import org.lwjgl.opengl.DisplayMode;
@@ -493,10 +494,11 @@
 
       if (BadlionClient.minecraft.method_237() != null && BadlionClient.minecraft.method_237().method_5256().equals(GuiType.CONNECTING)) {
          ConnectScreen var3 = (ConnectScreen)MinecraftClient.getInstance().currentScreen;
-         if (var3.getIP() != null && var3.getPort() != 0 && !var3.isConnecting()) {
-            var3.connect(var3.getIP(), var3.getPort());
+//         if (var3.getIP() != null && var3.getPort() != 0 && !var3.isConnecting()) {
+//            var3.connect(var3.getIP(), var3.getPort());
+//         }
+//         TODO: Mixin fix
          }
-      }
 
       BadlionClient.getBadlion().getAnalyicsManager().method_3511();
       BadlionClient.getBadlion().getFpsAverager().method_5527();
@@ -638,7 +640,7 @@
       return this.debugEnabled;
    }
 
-   public void handleFullScreen() {
+   public void handleFullScreen() throws LWJGLException {
       if (this.getActiveModProfile().getBetterframesConfig().getBorderlessWindow().isTrue()) {
          if (MinecraftClient.getInstance().isWindowFocused()) {
             System.setProperty("org.lwjgl.opengl.Window.undecorated", "true");
@@ -728,7 +730,9 @@
       BlockPos var2 = var1.method_3828();
       int var3 = Math.abs(((int)MinecraftClient.getInstance().player.x >> 4) - (var2.getX() >> 4));
       int var4 = Math.abs(((int)MinecraftClient.getInstance().player.z >> 4) - (var2.getZ() >> 4));
-      return MinecraftClient.SERVER_VIEW_DISTANCE == -1 || var3 <= MinecraftClient.SERVER_VIEW_DISTANCE && var4 <= MinecraftClient.SERVER_VIEW_DISTANCE;
+//      TODO: Mixin fix
+      return true;
+//      return MinecraftClient.SERVER_VIEW_DISTANCE == -1 || var3 <= MinecraftClient.SERVER_VIEW_DISTANCE && var4 <= MinecraftClient.SERVER_VIEW_DISTANCE;
    }
 
    public void closeActiveSlideoutSettingsPage() {
@@ -823,7 +827,7 @@
    }
 
    public void handleScreenshotCommand(String var1) {
-      class_674.handleCommand(var1);
+//      class_674.handleCommand(var1);
    }
 
    public String minecraftVersionString() {
