--- a/net\badlion\client\item\tag\Tag.java
+++ b/net\badlion\client\item\tag\Tag.java
@@ -1,6 +1,7 @@
 package net.badlion.client.item.tag;
 
 import java.io.DataOutput;
+import java.io.IOException;
 import java.util.Set;
 import net.badlion.Class_8;
 import net.badlion.Class_849;
@@ -74,7 +75,7 @@
         return this.field_2135.getKeys();
     }
 
-    public void write(String parameter_1, DataOutput parameter_2) {
+    public void write(String parameter_1, DataOutput parameter_2) throws IOException {
         CompoundTag.write(parameter_1, this.field_2135, parameter_2);
     }
 
