--- a/net/badlion/client/common/render/texture/AbstractTextureRegister.java
+++ b/net/badlion/client/common/render/texture/AbstractTextureRegister.java
@@ -32,15 +32,16 @@
       (new Thread(new Runnable() {
          public void run() {
             while(!Thread.currentThread().isInterrupted()) {
-               final AbstractTextureRegister.class_2117 var1;
+               AbstractTextureRegister.class_2117 var1;
                try {
                   while((var1 = (AbstractTextureRegister.class_2117)AbstractTextureRegister.this.field_5662.poll()) != null) {
                      try {
                         InputStream var2 = AbstractTextureRegister.class.getResourceAsStream("/assets/minecraft/" + var1.field_5666);
                         final BufferedImage var3 = ImageIO.read(var2);
+                        class_2117 finalVar = var1;
                         BadlionClient.minecraft.method_211(new Runnable() {
                            public void run() {
-                              AbstractTextureRegister.this.method_5783(var1.field_5665, var3, var1.field_5667, var1.field_5668);
+                              AbstractTextureRegister.this.method_5783(finalVar.field_5665, var3, finalVar.field_5667, finalVar.field_5668);
                            }
                         });
                      } catch (Exception var6) {
