--- a/net/badlion/client/common/gui/lib/element/BaseElement.java
+++ b/net/badlion/client/common/gui/lib/element/BaseElement.java
@@ -6,14 +6,14 @@
 import net.badlion.common.gui.lib.Size;
 
 public class BaseElement extends Element {
-   protected Size positionX = new Size();
-   protected Size positionY = new Size();
-   protected Size width = new Size();
-   protected Size height = new Size();
-   protected List children;
-   private List queue;
-   private BaseElement parent;
-   protected boolean iterating;
+   public Size positionX = new Size();
+   public Size positionY = new Size();
+   public Size width = new Size();
+   public Size height = new Size();
+   public List children;
+   public List queue;
+   public BaseElement parent;
+   public boolean iterating;
 
    public void method_3816() {
       this.field_2858 = this.positionX.method_3922();
