--- a/net\badlion\client\manager\GLManager.java
+++ b/net\badlion\client\manager\GLManager.java
@@ -90,7 +90,7 @@
 
     }
 
-    static LinkedList<GLManager.Matrix> getMatrixStack() {
+    public static LinkedList<GLManager.Matrix> getMatrixStack() {
         return matricies;
     }
 
@@ -130,21 +130,21 @@
     }
 
     public static class Class_2064 {
-        float field_5465;
-        private float field_5470;
-        private float field_5471;
-        private float field_5472;
-        private float field_5473;
-        float field_5466;
+        public float field_5465;
+        public float field_5470;
+        public float field_5471;
+        public float field_5472;
+        public float field_5473;
+        public float field_5466;
         private float field_5474;
         private float field_5475;
         private float field_5476;
         private float field_5477;
         private float field_5478;
         private float field_5479;
-        float field_5467;
-        float field_5468;
-        float field_5469;
+        public float field_5467;
+        public float field_5468;
+        public float field_5469;
         private float field_5480;
 
         private Class_2064() {
